packages/react-vanilla-components/src/components/CheckBox.tsx

```diff
@@ -55,10 +58,10 @@ const CheckBox = (props: PROPS) => {
             value={value}
             name={name}
             readOnly={readOnly}
-            disabled={!enabled}
+            disabled={context.disabled}
```

Comment:
What if checkbox is rendered without TnC?

Reply:
Removed this logic and implemented it via model, fetching the model of checkbox and enabling it after scroll.